pkg/tests/tasks/traffic/egress/egress_gateways_test.go

```diff
@@ -16,109 +16,267 @@ package egress
 
 import (
 	"fmt"
-	"strings"
 	"testing"
-	"time"
 
-	"github.com/maistra/maistra-test-tool/pkg/examples"
+	"github.com/maistra/maistra-test-tool/pkg/app"
 	"github.com/maistra/maistra-test-tool/pkg/util"
-	"github.com/maistra/maistra-test-tool/pkg/util/log"
-	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/common"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
-func cleanupEgressGateways() {
-	log.Log.Info("Cleanup")
-	sleep := examples.Sleep{Namespace: "bookinfo"}
-	util.KubeDeleteContents("bookinfo", util.RunTemplate(ExGatewayHTTPSTemplate, smcp))
-	util.KubeDeleteContents("bookinfo", util.RunTemplate(ExGatewayTemplate, smcp))
-	util.KubeDeleteContents("bookinfo", ExServiceEntryTLS)
-	util.KubeDeleteContents("bookinfo", ExServiceEntry)
-	sleep.Uninstall()
-	time.Sleep(time.Duration(20) * time.Second)
+func TestEgressGateways(t *testing.T) {
+	NewTest(t).Id("T13").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+
+		t.NewSubTest("HTTP").Run(func(t TestHelper) {
+			t.LogStep("Create a ServiceEntry to external istio.io")
+			oc.ApplyString(t, ns, ExServiceEntry)
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ExServiceEntry)
+			})
+			assertExternalHTTPRequestSuccessful(t, ns)
+
+			t.LogStep("Create a Gateway to external istio.io")
+			oc.ApplyTemplate(t, ns, ExGatewayTemplate, smcp)
+			t.Cleanup(func() {
```

Comment:
A question only, if you create two cleanup func are going to be executed in sequence when the test case finishes or is aborted by failure?

Reply:
All cleanup functions get executed (in reverse order).

Reply:
Ohhh, good to know that.